core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeParameterService.java

```diff
@@ -0,0 +1,18 @@
+package io.metadew.iesi.server.rest.componentTypes.dto;
+
+import io.metadew.iesi.metadata.definition.component.ComponentTypeParameter;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ComponentTypeParameterService implements IComponentTypeParameterService{
+    @Override
+    public ComponentTypeParameterDto convertToDto(ComponentTypeParameter componentTypeParameter, String name) {
+        return new ComponentTypeParameterDto(
+                name,
+                componentTypeParameter.getDescription(),
+                componentTypeParameter.getType(),
+                componentTypeParameter.isMandatory(),
+                componentTypeParameter.isEncrypted()
+        );
+    }
+}
```

Comment:
It is not possible to add resource assembler support for the ComponentTypeParameterDto.
The ComponentTypeParameter doesn't have the name getter, so I had to pass the nam as an argument.
Also,  the RepresentationModelAssemblerSupport doesn't provide the methods ToModel with a second argument.

So I let the service to convert the ComponentTypeParameter to Dto